plugins/org.jkiss.dbeaver.ui.editors.base/src/org/jkiss/dbeaver/ui/editors/EditorAccessibleAdapter.java

```diff
@@ -16,72 +16,36 @@
  */
 package org.jkiss.dbeaver.ui.editors;
 
-import org.eclipse.swt.accessibility.*;
+import org.eclipse.swt.accessibility.AccessibleControlAdapter;
+import org.eclipse.swt.accessibility.AccessibleControlEvent;
+import org.eclipse.swt.accessibility.AccessibleEvent;
+import org.eclipse.swt.accessibility.AccessibleListener;
 import org.eclipse.swt.custom.CTabFolder;
 import org.eclipse.swt.widgets.Composite;
-import org.eclipse.swt.widgets.TabFolder;
+import org.eclipse.swt.widgets.Control;
 import org.jkiss.code.NotNull;
 
-import java.lang.ref.WeakReference;
-import java.util.ArrayDeque;
-import java.util.Deque;
-
 public class EditorAccessibleAdapter extends AccessibleControlAdapter implements AccessibleListener {
-    private static final String DATA_KEY = EditorAccessibleAdapter.class.getName();
-
-    private final WeakReference<Composite> composite;
-    private boolean active;
-
-    private EditorAccessibleAdapter(@NotNull Composite composite) {
-        this.composite = new WeakReference<>(composite);
-    }
-
-    /**
-     * Checks if any control in the hierarchy has this accessible adapter installed and it's active.
-     * <p>
-     * In other words, if any method of this adapter was called at least once, it will be treated as "active".
-     */
-    public static boolean isActive(@NotNull Composite composite) {
-        for (Composite c = composite; c != null; c = c.getParent()) {
-            final Object data = c.getData(DATA_KEY);
-            if (data instanceof EditorAccessibleAdapter && ((EditorAccessibleAdapter) data).active) {
-                return true;
-            }
-        }
-
-        return false;
-    }
-
-    /**
-     * Installs this accessible adapter on the given control.
-     */
-    public static void install(@NotNull Composite composite) {
-        final EditorAccessibleAdapter adapter = new EditorAccessibleAdapter(composite);
 
-        final Accessible accessible = composite.getAccessible();
-        accessible.addAccessibleListener(adapter);
-        accessible.addAccessibleControlListener(adapter);
+    private static final String ACTIVE_NAME_EDITOR = "editor %s %s"; // $NON-NLS-0$
+    private static final String ACTIVE_NAME_TAB = "active tab %s %s of %s"; // $NON-NLS-0$
+    private final Control composite;
 
-        composite.setData(DATA_KEY, adapter);
+    EditorAccessibleAdapter(@NotNull Control composite) {
+        this.composite = composite;
```

Comment:
Are you sure it won't cause memory leaks? Theoretically, listeners are released when the control disposes, but anyway.

Reply:
Worth testing.

Reply:
Is there a particular reason why you've opted out of using a `WeakReference` here anyways?